fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -105,6 +105,7 @@ public abstract class IndexingBase {
     private boolean forceStampOverwrite = false;
     private final long startingTimeMillis;
     private long lastTypeStampCheckMillis;
+    private List<Index> luceneIndexes;
```

Comment:
I think a forward reference to an index type from one of the extension subsystems is an unnecessary violation of modularity, even if it doesn't call any actual code there.